packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -1,12 +1,350 @@
+     * @param depositId Unique ID corresponding to deposit order that caller wants to relay.
+     * @param depositTimestamp Timestamp of Deposit emitted by L2 contract when order was initiated.
+     * @param recipient Address on this network who should receive the relayed deposit.
+     * @param l1Token Token currency to pay recipient. This contract stores a mapping of
+     * `l1Token` to the canonical token currency on the L2 network that was deposited to the Deposit contract.
+     * @param amount Deposited amount.
+     * @param realizedFeePct Computed offchain by caller, considering the amount of available liquidity for the token
+     * currency needed to pay the recipient and the count of pending withdrawals at the `depositTimestamp`. This fee
+     * will be subtracted from the `amount`. If this value is computed incorrectly, then the relay can be disputed.
+     * @param maxFeePct Maximum fee that L2 Depositor can pay. `realizedFee` <= `maxFee`.
+     * @param proposerRewardPct Reward % of deposit amount to pay relayers.
+     */
+    function relayDeposit(
+        uint256 depositId,
+        uint256 depositTimestamp,
+        address recipient,
+        address l2Sender,
+        address l1Token,
+        uint256 amount,
+        // TODO: Allow caller to distinguish between slow and fast fees/rewards.
+        uint256 realizedFeePct,
+        uint256 maxFeePct,
+        uint256 proposerRewardPct
+    ) public {
+        // TODO: Should this perform a check that the amount to relay minus fees and rewards is <= current balance?
+        require(realizedFeePct <= maxFeePct, "Invalid realized fee");
+        Deposit storage newDeposit = deposits[depositId];
+        require(newDeposit.depositState == DepositState.Uninitialized, "Pending relay for deposit ID exists");
+        Deposit storage disputedDeposit = disputedDeposits[depositId][msg.sender];
+        require(
+            disputedDeposit.depositState == DepositState.Uninitialized,
+            "Pending dispute by relayer for deposit ID exists"
+        );
+
+        RelayAncillaryDataContents memory newRelayData =
+            RelayAncillaryDataContents({
+                depositId: depositId,
+                l2Sender: l2Sender,
+                recipient: recipient,
+                depositTimestamp: depositTimestamp,
+                l1Token: l1Token,
+                amount: amount,
+                maxFeePct: maxFeePct,
+                proposerRewardPct: proposerRewardPct,
+                realizedFeePct: realizedFeePct,
+                slowRelayer: msg.sender
+            });
+        bytes memory customAncillaryData = getRelayAncillaryData(newRelayData);
+
+        // Request a price for the relay identifier and propose "true" optimistically. These methods will pull the
+        // (proposer reward + proposer bond + final fee) from the caller.
+        // TODO: Figure out what to set for price request timestamp. I don't think we should simply set it to the
```

Comment:
I think the L1 block time is fine. that's what `getCurrentTime` on L1 will return and it means that from the proposal time (this method being called) + OO liveness for the bridging action to be verified.

Reply:
Cool that's how I have it now, will remove the comment